Guides/ExtendedJSON.md

```diff
@@ -0,0 +1,108 @@
+
+There are two types of Extended JSON: relaxed and canonical. 
+
+The `ExtendedJSONEncoder` and `ExtendedJSONDecoder` provide a way for any custom `Codable` classes to interact with 
+canonical or relaxed extended JSON.
+ 
+See the example below for an overview of working with these classes. 
+```swift
+struct Person: Codable, Equatable {
+    let name: String
+    let age: Int32
+}
+let bob = Person(name: "Bob", age: 25)
+
+// Try the canonical encoder
+let encoder = ExtendedJSONEncoder()
+encoder.mode = .canonical
+let canonicalEncoded = try encoder.encode(bob) // "{\"name\":\"Bob\",\"age\":{\"$numberInt\":\"25\"}}"
+
+// Try the relaxed encoder
+encoder.mode = .relaxed
+let relaxedEncoded = try encoder.encode(bob) // "{\"name\":\"Bob\",\"age\":25}}"
+
+// Try decoding the results
+let decoder = ExtendedJSONDecoder()
+let canonicalExtJSON = "{\"name\":\"Bob\",\"age\":{\"$numberInt\":\"25\"}}"
+let canonicalDecoded = try decoder.decode(Person.self, from: canonicalExtJSON.data(using: .utf8)!) // bob
+let relaxedExtJSON = "{\"name\":\"Bob\",\"age\":25}}"
+let relaxedDecoded = try decoder.decode(Person.self, from: relaxedExtJSON.data(using: .utf8)!) // bob
+// you decode from relaxed or canonical extended json and get the same result back (bob in this case)
+```
+Invalid input to both the `encode` and `decode` methods will result in error messages describing the invalid portion of 
+the input and what kind of input would be accepted in its place.
+
+### Relaxed and Canonical Extended JSON
+
+- _Relaxed Extended JSON_ - A string format based on the JSON standard that describes BSON documents. 
+Relaxed Extended JSON emphasizes readability and interoperability at the expense of type preservation.
+
+- _Canonical Extended JSON_ - A string format based on the JSON standard that describes BSON documents. 
+Canonical Extended JSON emphasizes type preservation at the expense of readability and interoperability.
+
+The example above illustrates the difference if you look at how the `age` field is represented in `canonicalExtJSON`
+versus how it is represented in `relaxedExtJSON`. 
+
+To see how all of the BSON types are represented in Canonical and Relaxed Extended JSON Format see this 
+[Conversion Table](https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#conversion-table).
+
+A thorough example Canonical Extended JSON document and its relaxed counterpart can be found 
+[here](https://github.com/mongodb/specifications/blob/master/source/extended-json.rst#canonical-extended-json-example).
+
+### Vapor
+If you are interested in using the `ExtendedJSONEncoder` and `ExtendedJSONDecoder` in your 
+[Vapor](https://docs.vapor.codes/4.0/) app, you can set them as the default encoder and decoder and thereby allow your 
+application to serialize data as ExtendedJSON, rather than the default plain JSON. 
+This way you can more easily interact with MongoDB and take advantage of the added type information.
```

Comment:
change to "This is recommended because not all BSON types currently support working with `JSONEncoder` and `JSONDecoder` and also so that you can take advantage of the added type information."